utils/data.py

```diff
@@ -47,8 +49,15 @@ def __init__(self, calendar, prices, sales, seq_len=1, horizon=1):
         self.prices = self._sell_prices(calendar, prices)
         self.sales = self._unit_sales(sales)
 
+        # normalize inputs to unit Gaussian distribution
+        self.day = (self.day - 1.4436644) / 6.093091
+        self.snap = (self.snap - 2.029751) / 3.42601
+        self.prices = (self.prices - 2.029751) / 3.42601
+        self.sales = (self.sales - 2.029751) / 3.42601
```

Comment:
Did you hardcode the mean and std because it takes a long time to compute?

Reply:
I wasn't sure about this either.

For the first competition, I also hardcoded it, because the normalization should be the same between the training, validation, and testing datasets. The validation and testing datasets do not have access to the samples of the training dataset, so then the options are either hardcoding or static variables. I opted for hardcoding, but if you prefer static variables, I can change it.

What I wasn't sure about is: Should I use the same mean and std for all the different types of data, or should I use a separate mean and std per type of data. (With type of data, I mean the 9 types of weekdays, weeks, monthdays, months, years, event types, SNAP, prices, and sales). The image data of the first competition was all the same type of data, so then it makes sense to use the same mean and std. But should I use the same mean and std for each type, or for two types like it is now, or have a separate mean and std for each of the 9 types of data?

Reply:
> For the first competition, I also hardcoded it, because the normalization should be the same between the training, validation, and testing datasets. The validation and testing datasets do not have access to the samples of the training dataset, so then the options are either hardcoding or static variables. I opted for hardcoding, but if you prefer static variables, I can change it.

Ah I see, static variables might make the code more readable, so that would have a slight preference for me.

> What I wasn't sure about is: Should I use the same mean and std for all the different types of data, or should I use a separate mean and std per type of data. (With type of data, I mean the 9 types of weekdays, weeks, monthdays, months, years, event types, SNAP, prices, and sales). The image data of the first competition was all the same type of data, so then it makes sense to use the same mean and std. But should I use the same mean and std for each type, or for two types like it is now, or have a separate mean and std for each of the 9 types of data?

A separate mean and std for the 9 types of data makes most sense to me. But `weekdays`, `years`, `months`, and `event_types` are all one-hot vectors so it doesn't really make sense to normalize those I think. Actually I don't think it is necessary to normalize any of the variables/features in `self.day` because these are not continuous. Not sure what kind of data `self.snap` is, but if it is just 0s and 1s then I don't think we need to normalize that either,